### PR TITLE
Prepare for 6.2.0 release

### PR DESCRIPTION
**Pull request by aucampia**
# Summary of changes

This should bring master branch to a point where we can tag and publish
6.2.0.

If there is anything I missed please let me know.

# Checklist

- [x] Checked that there aren't other open pull requests for
  the same change.
- [x] Checked that all tests and type checking passes.
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Comment by aucampia** [comment]
Got the list of merged PRs with:

```bash
gh search prs \
  --repo RDFLib/rdflib --merged --base master \
  --json assignees,author,authorAssociation,body,closedAt,commentsCount,createdAt,id,isLocked,isPullRequest,labels,number,repository,state,title,updatedAt,url \
  --limit 1000 --jq '[.[] | select(.closedAt >= "2021-12-19T00:00:00Z")]' \
  | jq '(. |= sort_by(.closedAt)) | reverse' > /var/tmp/merged-prs.json

jq -r '.[] | [ .url, .title ] | @tsv' /var/tmp/merged-prs.json | \
  head -n -3 | sort -r | \
  awk -F$'\t' '(match($1, "^.*/([^/]+)$", matches)){printf("* %s\n  [PR #%s](%s)\n", $2, matches[1], $1)}'
```

File: rdflib/__init__.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
 __docformat__ = "restructuredtext en"
 
 # The format of the __version__ line is matched by a regex in setup.py
-__version__ = "6.2.0-alpha"
+__version__ = "6.2.0"
 __date__ = "2021-12-20"
```
Comment (edmondchuc) [inline rdflib/__init__.py:49]: Should this date also be updated?
```suggestion
__date__ = "2022-12-20"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,43 +1,613 @@
...
-Type hints were added to the following modules:
-
-- `rdflib.namespace`: Nearly everything in this module has type hints now.
-
-PRs merged since last release:
-------------------------------
-
-* TODO
-
+# 2021-07-16 RELEASE 6.2.0
```
Comment (edmondchuc) [inline CHANGELOG.md:1]: Should this be 2022?
```suggestion
# 2022-07-16 RELEASE 6.2.0
```

**Review by edmondchuc — COMMENTED** [review]
This changelog really shows how much work you all put in to RDFLib! Great job and many thanks 👏

**Review by gjhiggins — APPROVED** [review]
lgtm

**Comment by aucampia** [comment]
> This changelog really shows how much work you all put in to RDFLib! Great job and many thanks clap

Thanks for suggestions also.

**Comment by aucampia** [comment]
Okay I published this to test pypi under https://test.pypi.org/project/rdflibx/ - as I don't have permissions to https://test.pypi.org/project/rdflib/ - @ashleysommer if you have a moment please add me there.

I also did some smoke tests and all seems fine.

Merging and tagging this.

**Comment by ajnelson-nist** [comment]
I look forward to being able to pull this in via PyPI!

**Comment by ajnelson-nist** [comment]
@aucampia - when will this be available via PyPI?

**Pull request merged**